src/main/service/publicationService.ts

```diff
@@ -65,34 +87,124 @@ export class PublicationService {
 
     private getCaseFromArtefact(artefact: Artefact, term: string, value: string): SearchObject {
         let foundObject: SearchObject = null;
-        artefact?.search.cases.forEach(singleCase => {
-            if (singleCase[term] == value) {
-                foundObject = singleCase;
-            }
-        });
+
+        if (artefact?.search.parties) {
+            artefact?.search.parties.forEach(party => {
+                const partyNames = this.constructPartyNames(party);
+                party.cases.forEach(singleCase => {
+                    if (singleCase[term] == value) {
+                        foundObject = { ...singleCase, partyNames };
+                    }
+                });
+            });
+        } else {
+            artefact?.search.cases.forEach(singleCase => {
+                if (singleCase[term] == value) {
+                    foundObject = { ...singleCase, partyNames: '' };
+                }
+            });
+        }
         return foundObject;
     }
 
     private getFuzzyCasesFromArtefact(artefacts: Artefact[], value: string): SearchObject[] {
         const matches: SearchObject[] = [];
         artefacts.forEach(artefact => {
-            artefact.search.cases.forEach(singleCase => {
-                if (singleCase.caseName && singleCase.caseName.toLowerCase().includes(value.toLowerCase())) {
-                    const alreadyExists = matches.find(
-                        m =>
-                            m.caseName === singleCase.caseName &&
-                            m.caseUrn === singleCase.caseUrn &&
-                            m.caseNumber === singleCase.caseNumber
-                    );
-                    if (!alreadyExists) {
-                        matches.push(singleCase);
+            if (artefact?.search.parties) {
+                artefact.search.parties.forEach(party => {
+                    const partyNames = this.constructPartyNames(party);
+                    party.cases.forEach(singleCase => {
+                        if (singleCase.caseName?.toLowerCase().includes(value.toLowerCase())) {
+                            this.storeUniquePartyCases(matches, singleCase, partyNames);
+                        }
+                    });
+                });
+            } else {
+                artefact.search.cases.forEach(singleCase => {
```

Comment:
Just double checking if this else is needed. Does the parties object only display in the search result if parties are present? If it always displays I don't believe the else would ever be triggered

Reply:
For artefacts uploaded before this PR is merged, they will not have any "parties" object so if we search for case name, we will need to consider the "cases" object. I guess this can be removed eventually if we know that all artefacts contain the "parties" search value but we need that now.

Reply:
Ah yes, good shout